dulwich/objects.py

```diff
@@ -2258,6 +2388,99 @@ def _get_extra(self) -> list[tuple[bytes, bytes | None]]:
     _TYPE_MAP[cls.type_num] = cls
 
 
+# Public API functions
+
+
+@replace_me(since="0.21.0", remove_in="0.24.0")
```

Comment:
I'm not sure I understand how this work. What should `parse_commit` be replaced with? The private `_parse_commit` function?